shared/src/androidMain/kotlin/org/hyperskill/app/main/presentation/MainViewModel.kt

```diff
@@ -1,8 +1,54 @@
 package org.hyperskill.app.main.presentation
 
+import android.os.Bundle
+import androidx.lifecycle.SavedStateHandle
+import com.chrynan.parcelable.core.Parcelable
+import com.chrynan.parcelable.core.decodeFromBundle
+import com.chrynan.parcelable.core.encodeToBundle
+import kotlinx.serialization.ExperimentalSerializationApi
+import org.hyperskill.app.main.injection.PlatformMainComponentImpl
+import org.hyperskill.app.main.presentation.AppFeature.Message.Initialize
 import ru.nobird.android.view.redux.viewmodel.ReduxViewModel
 import ru.nobird.app.presentation.redux.container.ReduxViewContainer
+import ru.nobird.app.presentation.redux.feature.Feature
 
+@OptIn(ExperimentalSerializationApi::class)
+/**
+ * Sent [AppFeature.Message.Initialize] only if the state was not saved before in [SavedStateHandle].
+ * If the state was saved, then [AppFeature] is initialized with the saved state.
+ * @see [PlatformMainComponentImpl], [MainViewModel.decodeState]
+ */
 class MainViewModel(
-    reduxViewContainer: ReduxViewContainer<AppFeature.State, AppFeature.Message, AppFeature.Action.ViewAction>
-) : ReduxViewModel<AppFeature.State, AppFeature.Message, AppFeature.Action.ViewAction>(reduxViewContainer)
\ No newline at end of file
+    reduxViewContainer: ReduxViewContainer<AppFeature.State, AppFeature.Message, AppFeature.Action.ViewAction>,
+    feature: Feature<AppFeature.State, AppFeature.Message, AppFeature.Action>,
+    savedStateHandle: SavedStateHandle
+) : ReduxViewModel<AppFeature.State, AppFeature.Message, AppFeature.Action.ViewAction>(reduxViewContainer) {
+    companion object {
+        private const val STATE_KEY = "AppFeatureState"
+
+        fun encodeState(state: AppFeature.State): Bundle =
+            Parcelable.Default.encodeToBundle(state, AppFeature.State.serializer())
+
+        /**
+         * Decodes state from [SavedStateHandle]
+         * @return [AppFeature.State] or null if state is not saved
+         */
+        fun decodeState(savedStateHandle: SavedStateHandle): AppFeature.State? {
+            val savedStateBundle: Bundle? = savedStateHandle[STATE_KEY]
+            return savedStateBundle?.let {
+                Parcelable.Default.decodeFromBundle(
+                    bundle = it,
+                    deserializer = AppFeature.State.serializer()
+                )
+            }
+        }
```

Comment:
Is it possible to test encode/decode here? It would be nice to have a test that checks all available `AppFeature.State`s (for example using `when`) and will fail on changes or when a new substate is added.

Reply:
@ivan-magda 
It's pretty hard to test  `encode` & `decode` methods because internally they use the Android Parcelable class, which is not accessible without an Android device or emulator. That's why I've just added a serialization test to check, that all subclasses are Serializable.